### PR TITLE
Fix part #22: Navigation drawer part 1

### PR DESCRIPTION
**Pull request by nikitamarysolomanpvt**
## Explanation
duplicate of #84
xml files in Navigation drawer
as per design document
https://docs.google.com/document/d/1Q9957ri3TlsTWdwBVAjn4g9EJzqmzBqrSXCHI3yRTEE/edit?usp=sharing

## Mock
https://xd.adobe.com/spec/e2239cf4-9cde-4c08-5296-25316c1f0a14-9412/screen/346dd03d-464b-4b49-9d7f-7ec6264fefd6/TP-Play-Side-Menu-1

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: app/src/main/res/drawable/ic_admin_settings_icon_grey_24dp.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+  android:width="24dp"
+  android:height="24dp"
+  android:viewportWidth="200"
+  android:viewportHeight="200">
+  <path
+    android:fillColor="#707070"
```
Comment (rt4914) [inline app/src/main/res/drawable/ic_admin_settings_icon_grey_24dp.xml]: @mschanteltc @nikitamarysolomanpvt There is inconsistency with icon color:
 Icon color for `My Downloads` and `Profile Chooser` is `#707070` and the icon color for `Options` and `Help` is `#727272`, we needs these colors to be consistent.

File: app/src/main/res/values/dimens.xml [hunk 2]
```diff
@@ -1,5 +1,15 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
+  <!-- Default screen margins, per the Android Design guidelines. -->
+  <dimen name="activity_horizontal_margin">16dp</dimen>
+  <dimen name="activity_vertical_margin">16dp</dimen>
+  <dimen name="nav_header_vertical_spacing">8dp</dimen>
+  <dimen name="nav_header_height">176dp</dimen>
+  <dimen name="margin_30">30dp</dimen>
+  <dimen name="image_dim_24">24dp</dimen>
+  <dimen name="margin_10">10dp</dimen>
```
Comment (rt4914) [inline app/src/main/res/values/dimens.xml]: The margin/padding name ending with a fix value means that it won't change, and the values which do not change at all it is preferable not to declare them in `dimens.xml` file.
Example: `activity_horizontal_margin ` is fine to use but `margin_30` is not.

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -1,6 +1,16 @@
...
+  <string name="nav_header_title">Sean</string>
+  <string name="nav_header_subtitle">14 Stories Completed | 3 Topics In Progress</string>
+  <string name="nav_header_desc">Navigation header</string>
+  <string name="menu_home">Home</string>
+  <string name="menu_help">Help</string>
+  <string name="admin_settings">Admin Settings</string>
+  <!-- TODO: Remove or change this placeholder text -->
+  <string name="drawer_open">Open</string>
+  <string name="drawer_close">Close</string>
+  <!-- TODO: Remove or change this placeholder text -->
```
Comment (rt4914) [inline app/src/main/res/values/strings.xml]: TODO's should have an issue or a developer name associated with them. So file an issue and then link that issue here.

**Review by rt4914 — CHANGES_REQUESTED** [review]
I am getting this error on Google Pixel 3 (Real device) and Galaxy Nexus (Emulator).
Screenshot 2019-11-25 at 10 34 49 AM

File: app/src/main/res/anim/slide_out_left.xml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<set xmlns:android="http://schemas.android.com/apk/res/android">
+  <translate android:fromXDelta="0"
+    android:toXDelta="-100%p"
```
Comment (veena14cs) [inline app/src/main/res/anim/slide_out_left.xml]: Please check this.

File: app/src/main/res/anim/slide_right_in.xml [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<set xmlns:android="http://schemas.android.com/apk/res/android">
+  <translate
+    android:duration="@integer/duration_anim_300_ms"
+    android:fromXDelta="100%p"
```
Comment (veena14cs) [inline app/src/main/res/anim/slide_right_in.xml]: Ditto here.

File: app/src/main/res/layout/help_activity.xml [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  android:id="@+id/drawer_layout"
+  android:layout_width="match_parent"
+  android:layout_height="match_parent">
+
+  <LinearLayout
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical">
+
+    <LinearLayout
```
Comment (veena14cs) [inline app/src/main/res/layout/help_activity.xml]: I think you can remove this extra layout, and include toolbar in the above layout.

File: .idea/gradle.xml [hunk 7]
```diff
@@ -1,22 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (jamesxu0) [inline .idea/gradle.xml]: I think it's better to not delete this file.

File: app/src/main/res/drawable/ic_admin_settings_icon_grey_24dp.xml (hunk 1 above)
Comment (nikitamarysolomanpvt) [reply app/src/main/res/drawable/ic_admin_settings_icon_grey_24dp.xml]: i was using #707070 as in mock color codes  were not consistent.

File: app/src/main/res/values/dimens.xml (hunk 2 above)
Comment (nikitamarysolomanpvt) [reply app/src/main/res/values/dimens.xml]: activity_horizontal_margin is already there

Comment (nikitamarysolomanpvt) [reply app/src/main/res/values/dimens.xml]: Removed

File: app/src/main/res/anim/slide_out_left.xml (hunk 4 above)
Comment (nikitamarysolomanpvt) [reply app/src/main/res/anim/slide_out_left.xml]: changed to -100%

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (nikitamarysolomanpvt) [reply app/src/main/res/values/strings.xml]: Removed

File: app/src/main/res/anim/slide_right_in.xml (hunk 5 above)
Comment (nikitamarysolomanpvt) [reply app/src/main/res/anim/slide_right_in.xml]: Done

File: app/src/main/res/layout/help_activity.xml (hunk 6 above)
Comment (nikitamarysolomanpvt) [reply app/src/main/res/layout/help_activity.xml]: Done

File: app/src/main/res/layout/fragment_drawer.xml [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+    android:layout_height="wrap_content"
+    android:layout_gravity="bottom"
+    android:orientation="horizontal"
+    android:padding="16dp">
+
+    <ImageView
+      android:layout_width="@dimen/image_dim_24"
+      android:layout_height="@dimen/image_dim_24"
+      android:layout_gravity="center_vertical"
+      android:layout_margin="10dp"
```
Comment (veena14cs) [inline app/src/main/res/layout/fragment_drawer.xml]: Please make this multiple of 4.

**Comment by nikitamarysolomanpvt** [comment]
> I am getting this error on Google Pixel 3 (Real device) and Galaxy Nexus (Emulator).
> Screenshot 2019-11-25 at 10 34 49 AM

Reverted home_activity.xml . Would add it in next PR as it is having dependency.

File: app/src/main/res/layout/fragment_drawer.xml [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+      android:src="@drawable/ic_admin_settings_icon_grey_24dp" />
+
+    <TextView
+      android:layout_width="match_parent"
+      android:layout_height="wrap_content"
+      android:layout_gravity="center_vertical"
+      android:layout_marginLeft="30dp"
```
Comment (veena14cs) [inline app/src/main/res/layout/fragment_drawer.xml]: Use margin start and margin end to support RTL.

File: app/src/main/res/layout/nav_header_navigation_drawer.xml [hunk 10]
```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+  xmlns:app="http://schemas.android.com/apk/res-auto"
+  android:layout_width="match_parent"
+  android:layout_height="@dimen/nav_header_height"
+  android:background="@drawable/side_nav_bar"
+  android:gravity="bottom"
+  android:orientation="vertical"
+  android:paddingLeft="@dimen/activity_horizontal_margin"
```
Comment (veena14cs) [inline app/src/main/res/layout/nav_header_navigation_drawer.xml]: use paddingStart.

**Review by veena14cs — CHANGES_REQUESTED** [review]
Please do the changes and assign me back.

File: app/src/main/res/drawable/ic_admin_settings_icon_grey_24dp.xml (hunk 1 above)
Comment (rt4914) [reply app/src/main/res/drawable/ic_admin_settings_icon_grey_24dp.xml]: @nikitamarysolomanpvt Check this link: https://xd.adobe.com/spec/e2239cf4-9cde-4c08-5296-25316c1f0a14-9412/screen/346dd03d-464b-4b49-9d7f-7ec6264fefd6/TP-Play-Side-Menu-1/

File: app/src/main/res/anim/slide_out_left.xml [hunk 11]
```diff
@@ -0,0 +1,6 @@
+<set xmlns:android="http://schemas.android.com/apk/res/android">
+  <translate android:fromXDelta="0"
+    android:toXDelta="-100%"
+    android:duration="@integer/duration_anim_300_ms"/>
```
Comment (rt4914) [inline app/src/main/res/anim/slide_out_left.xml]: Space issue: ` />`
Suggest using `Option+cmd+L` in all files to automatically resolve everything.

File: app/src/main/res/layout/fragment_drawer.xml [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+  <LinearLayout
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:layout_gravity="bottom"
+    android:orientation="horizontal"
+    android:padding="16dp">
+
+    <ImageView
+      android:layout_width="@dimen/image_dim_24"
+      android:layout_height="@dimen/image_dim_24"
```
Comment (rt4914) [inline app/src/main/res/layout/fragment_drawer.xml]: If it is fix 24dp then suggest declaring it directly here and do not declare this in dimens file.

File: app/src/main/res/layout/fragment_drawer.xml (hunk 8 above)
Comment (rt4914) [reply app/src/main/res/layout/fragment_drawer.xml]: +1

File: app/src/main/res/layout/fragment_drawer.xml (hunk 9 above)
Comment (rt4914) [reply app/src/main/res/layout/fragment_drawer.xml]: +1

Comment (rt4914) [reply app/src/main/res/layout/fragment_drawer.xml]: Also make it either 28 or 32dp.

File: app/src/main/res/layout/help_activity.xml [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+    <FrameLayout
+      android:id="@+id/container_body"
+      android:layout_width="match_parent"
+      android:layout_height="0dp"
+      android:layout_weight="1" />
+  </LinearLayout>
+
+  <fragment
+    android:id="@+id/fragment_navigation_drawer"
+    android:name="org.oppia.app.drawer.NavigationDrawerFragment"
```
Comment (rt4914) [inline app/src/main/res/layout/help_activity.xml]: This file is not present currently.

File: app/src/main/res/drawable/ic_admin_settings_icon_grey_24dp.xml (hunk 1 above)
Comment (nikitamarysolomanpvt) [reply app/src/main/res/drawable/ic_admin_settings_icon_grey_24dp.xml]: Ya the same iam speaking different icons have different color code

File: app/src/main/res/layout/help_activity.xml [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+
+    <include
+      android:id="@+id/toolbar"
+      layout="@layout/toolbar" />
+
+    <FrameLayout
+      android:id="@+id/container_body"
+      android:layout_width="match_parent"
+      android:layout_height="0dp"
+      android:layout_weight="1" />
```
Comment (rt4914) [inline app/src/main/res/layout/help_activity.xml]: Remove this and suggest using `match_parent` in `layout_height`

File: app/src/main/res/menu/navigation_drawer_menu.xml [hunk 15]
```diff
@@ -0,0 +1,16 @@
...
+  <group android:checkableBehavior="single">
+    <item
+      android:id="@+id/nav_home"
+      android:icon="@drawable/ic_home_icon_grey_24dp"
+      android:title="@string/menu_home" />
+    <item
+      android:id="@+id/nav_help"
+      android:icon="@drawable/ic_help_grey_icon_24dp"
+      android:title="@string/menu_help"
+      android:visible="true" />
```
Comment (rt4914) [inline app/src/main/res/menu/navigation_drawer_menu.xml]: Why is `nav_help` visible and in `nav_home` it has not been defined?

**Review by rt4914 — CHANGES_REQUESTED** [review]
PTAL

File: .idea/gradle.xml (hunk 7 above)
Comment (nikitamarysolomanpvt) [reply .idea/gradle.xml]: Done

File: app/src/main/res/layout/fragment_drawer.xml (hunk 8 above)
Comment (nikitamarysolomanpvt) [reply app/src/main/res/layout/fragment_drawer.xml]: Done

File: app/src/main/res/layout/fragment_drawer.xml (hunk 9 above)
Comment (nikitamarysolomanpvt) [reply app/src/main/res/layout/fragment_drawer.xml]: Done

File: app/src/main/res/layout/nav_header_navigation_drawer.xml (hunk 10 above)
Comment (nikitamarysolomanpvt) [reply app/src/main/res/layout/nav_header_navigation_drawer.xml]: Done

File: app/src/main/res/layout/fragment_drawer.xml (hunk 12 above)
Comment (nikitamarysolomanpvt) [reply app/src/main/res/layout/fragment_drawer.xml]: Done

File: app/src/main/res/anim/slide_out_left.xml (hunk 11 above)
Comment (nikitamarysolomanpvt) [reply app/src/main/res/anim/slide_out_left.xml]: Done

File: app/src/main/res/drawable/ic_admin_settings_icon_grey_24dp.xml (hunk 1 above)
Comment (mschanteltc) [reply app/src/main/res/drawable/ic_admin_settings_icon_grey_24dp.xml]: The colors of the icons should be #727272. I updated the mocks so the colors are consistent with one another

Comment (nikitamarysolomanpvt) [reply app/src/main/res/drawable/ic_admin_settings_icon_grey_24dp.xml]: Done

File: app/src/main/res/layout/help_activity.xml (hunk 13 above)
Comment (nikitamarysolomanpvt) [reply app/src/main/res/layout/help_activity.xml]: removed help_activity.xml due to dependency

File: app/src/main/res/layout/help_activity.xml (hunk 14 above)
Comment (nikitamarysolomanpvt) [reply app/src/main/res/layout/help_activity.xml]: removed help_activity.xml due to dependency.

File: app/src/main/res/menu/navigation_drawer_menu.xml (hunk 15 above)
Comment (nikitamarysolomanpvt) [reply app/src/main/res/menu/navigation_drawer_menu.xml]: Removed

**Review by rt4914 — APPROVED** [review]
LGTM

File: app/src/main/res/layout/nav_header_navigation_drawer.xml [hunk 16]
```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+  xmlns:app="http://schemas.android.com/apk/res-auto"
+  android:layout_width="match_parent"
+  android:layout_height="@dimen/nav_header_height"
```
Comment (veena14cs) [inline app/src/main/res/layout/nav_header_navigation_drawer.xml]: If you are not using this anywhere you can directly specify height.

**Review by veena14cs — APPROVED** [review]
LGTM one nit change.

File: app/src/main/res/layout/nav_header_navigation_drawer.xml (hunk 16 above)
Comment (nikitamarysolomanpvt) [reply app/src/main/res/layout/nav_header_navigation_drawer.xml]: Done

**Pull request merged**